plugins/code-climate/src/plugin.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2020 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ProductionCodeClimateApi, codeClimateApiRef } from './api';
+import {
+  createApiFactory,
+  createPlugin,
+  createRouteRef,
+  discoveryApiRef,
+  identityApiRef,
+  createComponentExtension,
+  createRoutableExtension,
+} from '@backstage/core-plugin-api';
+import { Entity } from '@backstage/catalog-model';
+
+export const CODECLIMATE_REPO_ID_ANNOTATION = 'codeclimate.com/repo-id';
+
+export const useProjectSlug = (entity: Entity) => {
+  return entity?.metadata.annotations?.[CODECLIMATE_REPO_ID_ANNOTATION] ?? '';
+};
+
+export const rootRouteRef = createRouteRef({
+  id: 'code-climate',
+});
+
+export const codeClimatePlugin = createPlugin({
+  id: 'code-climate',
+  apis: [
+    createApiFactory({
+      api: codeClimateApiRef,
+      deps: {
+        discoveryApi: discoveryApiRef,
+        identityApi: identityApiRef,
+      },
+      factory: ({ discoveryApi, identityApi }) =>
+        new ProductionCodeClimateApi(discoveryApi, identityApi),
+    }),
+  ],
+  routes: {
+    root: rootRouteRef,
+  },
+});
+
+export const EntityCodeClimateContent = codeClimatePlugin.provide(
+  createRoutableExtension({
+    name: 'EntityCodeClimateContent',
+    mountPoint: rootRouteRef,
+    component: () =>
+      import('./components/CodeClimateWidget').then(m => m.CodeClimateWidget),
```

Comment:
does this one really work as a routable full-dedicated-page thing? i thought it was exclusively just a card